where_to_go/lib/main.dart

```diff
@@ -1,44 +1,54 @@
-// zad 4
-// import "package:flutter/material.dart";
-// import "package:hooks_riverpod/hooks_riverpod.dart";
-
-// import "utils/app_router.dart";
-
-// void main() {
-//   runApp(const ProviderScope(child: TravelApp()));
-// }
-
-// class TravelApp extends StatelessWidget {
-//   const TravelApp({super.key});
-
-//   @override
-//   Widget build(BuildContext context) {
-//     return MaterialApp.router(
-//       routerConfig: goRouter,
-//       title: "Travel App",
-//     );
-//   }
-// }
-
-// zad2
 import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
 
-import "features/places/places_inherited.dart";
-import "utils/app_router_inherited.dart";
+import "features/theme/theme_provider.dart";
+import "themes/app_theme.dart";
+import "themes/enums/mode_theme.dart";
+import "themes/utils.dart";
+import "utils/app_router.dart";
+import "utils/database/providers/database_provider.dart";
 
 void main() {
-  runApp(const TravelApp());
+  WidgetsFlutterBinding.ensureInitialized();
+
+  runApp(const ProviderScope(child: TravelApp()));
 }
 
-class TravelApp extends StatelessWidget {
+class TravelApp extends ConsumerWidget {
   const TravelApp({super.key});
 
   @override
-  Widget build(BuildContext context) {
-    return PlacesInherited(
-        child: MaterialApp.router(
-      routerConfig: goRouterInherited,
-      title: "Travel App",
-    ));
+  Widget build(BuildContext context, WidgetRef ref) {
+    final db = ref.watch(databaseProvider);
+    final baseTheme = _getTheme(context, ref);
+
+    return db.when(
+      data: (_) => MaterialApp.router(
+        routerConfig: goRouter,
+        title: "Travel App",
+        theme: baseTheme,
+      ),
+      error: (error, stack) => const Center(child: Text("Blad")),
+      loading: () => const Center(child: CircularProgressIndicator()),
+    );
+  }
+
+  ThemeData _getTheme(BuildContext context, WidgetRef ref) {
+    final appTheme = AppTheme();
+    final themeAsync = ref.watch(themeNotifierProvider);
+    final modeTheme = themeAsync.valueOrNull ?? ModeTheme.system;
+
+    ThemeData baseTheme = appTheme.light;
+
+    switch (modeTheme) {
+      case ModeTheme.light:
+        baseTheme = appTheme.light;
+      case ModeTheme.dark:
+        baseTheme = appTheme.dark;
+      case ModeTheme.system:
+        baseTheme = context.isDarkMode ? appTheme.dark : appTheme.light;
+    }
+
```

Comment:
to jest nitpicking ale polecam nowy syntax switcha w dartcie. Mniej więcej tak to wygląda (piszę z głowy więc mogłem się pomylić kapkę):
```suggestion
 final baseTheme = switch (modeTheme) {
     ModeTheme.light => appTheme.light,
     ModeTheme.dark => appTheme.dark,
     ModeTheme.system => context.isDarkMode ? appTheme.dark : appTheme.light,
    };
```